src/Controller/Admin/BuildingsController.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Controller\Admin;
+
+use App\Entity\Building;
+use App\Form\Type\BuildingType;
+use App\Form\Type\DeleteCancelType;
+use App\Repository\BuildingRepository;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+class BuildingsController extends AbstractController
+{
+    private BuildingRepository $buildingRepository;
+
+    private EntityManagerInterface $entityManager;
+
+    public function __construct(BuildingRepository $buildingRepository, EntityManagerInterface $entityManager)
+    {
+        $this->buildingRepository = $buildingRepository;
+        $this->entityManager = $entityManager;
+    }
+
+
+    /**
+     * @Route("/admin/buildings", name="app_admin_buildings", methods={"GET"})
+     */
+    public function showBuildingsAction(): Response
+    {
+        $buildings = $this->buildingRepository->findAll();
+
+        return $this->render('admin/buildings/index.html.twig', [
+            'buildings' => $buildings
+        ]);
+    }
+
+    /**
+     * @Route ("admin/buildings/add", name="app_admin_buildings_add", methods={"GET", "POST"})
```

Comment:
you don't need the `/add` here and you can also call it just `/building` without the s

Reply:
I have refactorized all routes and route names in admin area. My console debug:router is:
```
  admin_buildings                                GET        ANY      ANY    /admin/buildings                   
  admin_buildings_add                            GET|POST   ANY      ANY    /admin/buildings/add               
  admin_buildings_delete                         GET|POST   ANY      ANY    /admin/buildings/delete/{id}       
  admin_buildings_update                         GET|POST   ANY      ANY    /admin/buildings/update/{id}       
  admin_login                                    GET|POST   ANY      ANY    /admin/login                       
  admin_logout                                   ANY        ANY      ANY    /admin/logout                      
  admin                                          GET        ANY      ANY    /admin                             
  admin_programme                                GET        ANY      ANY    /admin/programmes                  
  admin_programme_add                            GET|POST   ANY      ANY    /admin/programmes/add              
  admin_programme_delete                         GET|POST   ANY      ANY    /admin/programmes/delete/{id}      
  admin_programme_update                         GET|POST   ANY      ANY    /admin/programmes/update/{id}      
  admin_rooms                                    GET        ANY      ANY    /admin/rooms                       
  admin_rooms_add                                GET|POST   ANY      ANY    /admin/rooms/add                   
  admin_rooms_delete                             GET|POST   ANY      ANY    /admin/rooms/delete/{id}           
  admin_rooms_update                             GET|POST   ANY      ANY    /admin/rooms/update/{id}           
  admin_users                                    GET        ANY      ANY    /admin/users                       
  admin_users_add                                GET|POST   ANY      ANY    /admin/users/add                   
  admin_users_delete                             GET|POST   ANY      ANY    /admin/users/delete/{id}           
  admin_users_update                             GET|POST   ANY      ANY    /admin/users/update/{id}   
```